internal/pkg/template/override/override.go

```diff
@@ -0,0 +1,39 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package override renders the manifest override rules to the CloudFormation template.
+package override
+
+// OverrideCloudFormationTemplate overrides the given CloudFormation template by applying
+// the override rules.
+func OverrideCloudFormationTemplate(overrideRules []string, origTemp string) (string, error) {
+	content, err := unmarshalCFNYaml(origTemp)
```

Comment:
Would it be a good idea to have
```go
type Template []byte
```
and then have `Template` implements the YAML `Unmarshaler` interface by doing
```go
func (t *Template) UnmarshalYAML
```
We can also do 
```go
func (t *Template) ApplyTemplateOverride(rules []Rule) (*Template, error)
```

Reply:
i think it might be more useful to make the unmarshalled template content to be a type...open to ideas tho